blazar/plugins/oshosts/host_plugin.py

```diff
@@ -987,28 +987,41 @@ def poll_resource_failures(self):
                 reservation = db_utils.get_most_recent_reservation_info_by_host_id(host['id'])
                 # ignore the reservation which is active, as the host must already be in the right pool
                 if reservation and reservation["reservation_status"] == status.reservation.ACTIVE:
-                    LOG.info(f"{host['hypervisor_hostname']} is in a active reservation"
-                             f"{reservation['reservation_id']}")
+                    LOG.debug(f"{host['hypervisor_hostname']} is in a active reservation"
+                             f" {reservation['reservation_id']} - skipping aggregate clean up")
                     continue
                 # host not in 'active' or 'pending' reservation should be in freepool
                 host_uuid = host['hypervisor_hostname']
                 curr_agg = host_to_agg_map.get(host['hypervisor_hostname'])
-                if curr_agg and (curr_agg.name == freepool.name):
+                if not curr_agg:
+                    LOG.warning(f"f{host['hypervisor_hostname']} not found in any aggregate - moving to freepool")
+                    # if the host is not in any aggregate and in non-active reservation
+                    # it should be moved to freepool
+                    freepool.add_host(host_uuid)
+                    continue
+                if curr_agg.name == freepool.name:
                     # if the host is already in freepool skip it
-                    LOG.info(f"{host['hypervisor_hostname']} is already in a freepool - skipping it")
+                    LOG.debug(f"{host['hypervisor_hostname']} is already in a freepool - skipping aggregate clean up")
```

Comment:
Do we need this case? Isn't this just the correct behavior?

Reply:
We would like to do nothing if the host is already in a freepool.

Reply:
this is here because we depend on the `continue` for these cases, which I admit is a bit confusing